### PR TITLE
Add README and literature background to treebank

### PR DESCRIPTION
**Pull request by lilianabs**
Reference Issue [340](https://github.com/FluxML/model-zoo/issues/340). 

Following some of the comments, this example contains fewer explanations and instead references the original paper that explains in detail how to model works.

### CODE REVIEW

File: text/treebank/recursive.jl [hunk 1]
```diff
@@ -1,25 +1,63 @@
...
+
+    ## Load data
     @info("Loading Data...")
     train_data, alphabet = getdata()    
 
     @info("Constructing model....")
     embedding = randn(Float32, args.N, length(alphabet))
 
+    @info("Size of the embegging")
+    @info(size(embedding))
```
Comment (ToucheSir) [inline text/treebank/recursive.jl:59]:
```suggestion
    @info "Size of the embedding" size(embedding)
```

File: text/treebank/recursive.jl [hunk 2]
```diff
@@ -1,25 +1,63 @@
...
 include("data.jl")
 
+# We set default values for the hyperparameters:
+
 @with_kw mutable struct Args
-    lr::Float64 = 1e-3    # Learning rate
+    lr::Float64 = 1e-3    ## Learning rate
     N::Int = 300
-    throttle::Int = 10    # Throttle timeout
+    throttle::Int = 10    ## Throttle timeout
```
Comment (ToucheSir) [inline text/treebank/recursive.jl:37]: Are these `##` special literate syntax as well?

Comment (lilianabs) [reply text/treebank/recursive.jl:37]: Yes, they're special literate syntax :)

**Review by ToucheSir — APPROVED** [review]
Thanks! If there's nothing else you'd like to add, this is ready to merge.

**Pull request merged**